src/Components/WelcomePage/WelcomePage.js

```diff
@@ -1,8 +1,30 @@
-import React from "react";
-import { Link } from 'react-router-dom'
+import React, {useEffect} from "react";
+import { Link, useHistory } from 'react-router-dom'
+import { getRandomWorldData } from "../../apiCalls";
+import { useSelector, useDispatch } from 'react-redux';
+import { getDiscoveredWorlds } from '../rootSlice';
 import './WelcomePage.css'
 
 export const WelcomePage = () => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const displayedWorlds = useSelector((state) => state.root.discoveredWorlds)
+
+  const discoverNewWorld = ()=> {
+    getRandomWorldData()
+    .then(data=> {
+      const addOne = [...displayedWorlds, data]
+      dispatch(getDiscoveredWorlds(addOne))
+      // console.log(data.id)
+      history.push(`/world/${data.id}`)
```

Comment:
can we discuss further the use of the useHistory method? I am on board with the dispatch of getDiscoveredWorlds to eventually update the discoveredWorlds global state, but why not just make the button a navlink to the path of world/${data.id}? we can discuss more in person!

Reply:
We can definitely discuss it further - i only resorted to using the useHistory way when i failed to make a <Redirect /> router component aim at that same sort of interpolated path

If we can make it work with a navlink, link, redirect, etc, i am down to switch it up!

Reply:
yeah i agree with priscilla here. i think a navlink would work better